0.3.27 fails to compile with `NO_CBLAS = 1`

Please see the following two workflows.

https://github.com/TLCFEM/prebuilds/actions/runs/8563626084/workflow#L21

https://github.com/TLCFEM/prebuilds/actions/runs/8563364136/workflow#L21

Here's the error I encountered:

```
cc -O2 -pipe -g -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer  -O2 -DMAX_STACK_ALLOC=2048 -Wall -m64 -DF_INTERFACE_GFORT -fPIC -DDYNAMIC_ARCH -DDYNAMIC_OLDER -DNO_LAPACKE -DSMP_SERVER -DNO_WARMUP -DMAX_CPU_NUMBER=64 -DMAX_PARALLEL_NUMBER=1 -DBUILD_SINGLE=1 -DBUILD_DOUBLE=1 -DBUILD_COMPLEX=1 -DBUILD_COMPLEX16=1 -DVERSION=\"0.3.27\" -UASMNAME -UASMFNAME -UNAME -UCNAME -UCHAR_NAME -UCHAR_CNAME -DASMNAME= -DASMFNAME=_ -DNAME=_ -DCNAME= -DCHAR_NAME=\"_\" -DCHAR_CNAME=\"\" -DNO_AFFINITY -I. -O2 -DSMALL_MATRIX_OPT -DUTEST_CHECK -DSANITY_CHECK -DREFNAME=f_ -DMAX_STACK_ALLOC=2048 -Wall -m64 -DF_INTERFACE_GFORT -fPIC -DDYNAMIC_ARCH -DDYNAMIC_OLDER -DNO_LAPACKE -DSMP_SERVER -DNO_WARMUP -DMAX_CPU_NUMBER=64 -DMAX_PARALLEL_NUMBER=1 -DBUILD_SINGLE=1 -DBUILD_DOUBLE=1 -DBUILD_COMPLEX=1 -DBUILD_COMPLEX16=1 -DVERSION=\"0.3.27\" -UASMNAME -UASMFNAME -UNAME -UCNAME -UCHAR_NAME -UCHAR_CNAME -DASMNAME= -DASMFNAME=_ -DNAME=_ -DCNAME= -DCHAR_NAME=\"_\" -DCHAR_CNAME=\"\" -DNO_AFFINITY -I.. -L/app/lib -Wl,-z,relro,-z,now -Wl,--as-needed  -o openblas_utest utest_main.o test_min.o test_amax.o test_ismin.o test_rotmg.o test_axpy.o test_dotu.o test_dsdot.o test_swap.o test_rot.o test_dnrm2.o test_zscal.o test_amin.o test_axpby.o test_potrs.o test_fork.o test_post_fork.o ../libopenblasp-r0.3.27.a -lm -lpthread -lgfortran -lm -lpthread -lgfortran -L/usr/lib/gcc/x86_64-unknown-linux-gnu/13.2.0 -L/usr/lib/gcc/x86_64-unknown-linux-gnu/13.2.0/../../../x86_64-linux-gnu -L/usr/lib/gcc/x86_64-unknown-linux-gnu/13.2.0/../../../../lib64 -L/lib/x86_64-linux-gnu -L/lib/../lib64 -L/usr/lib/x86_64-linux-gnu -L/usr/lib/../lib64 -L/usr/lib/gcc/x86_64-unknown-linux-gnu/13.2.0/../../../../x86_64-unknown-linux-gnu/lib -L/usr/lib/gcc/x86_64-unknown-linux-gnu/13.2.0/../../..  -lgfortran -lm -lquadmath -lm -lc 
/usr/lib/gcc/x86_64-unknown-linux-gnu/13.2.0/../../../../x86_64-unknown-linux-gnu/bin/ld: test_zscal.o: in function `__ctest_zscal_i_nan_run':
/run/build/openblas/utest/test_zscal.c:16:(.text+0xc2): undefined reference to `cblas_zscal'
/usr/lib/gcc/x86_64-unknown-linux-gnu/13.2.0/../../../../x86_64-unknown-linux-gnu/bin/ld: test_zscal.o: in function `__ctest_zscal_i_nan_inc_2_run':
/run/build/openblas/utest/test_zscal.c:28:(.text+0x1fc): undefined reference to `cblas_zscal'
/usr/lib/gcc/x86_64-unknown-linux-gnu/13.2.0/../../../../x86_64-unknown-linux-gnu/bin/ld: test_zscal.o: in function `__ctest_zscal_nan_i_run':
/run/build/openblas/utest/test_zscal.c:39:(.text+0x362): undefined reference to `cblas_zscal'
/usr/lib/gcc/x86_64-unknown-linux-gnu/13.2.0/../../../../x86_64-unknown-linux-gnu/bin/ld: test_zscal.o: in function `__ctest_zscal_nan_i_inc_2_run':
/run/build/openblas/utest/test_zscal.c:51:(.text+0x4a3): undefined reference to `cblas_zscal'
/usr/lib/gcc/x86_64-unknown-linux-gnu/13.2.0/../../../../x86_64-unknown-linux-gnu/bin/ld: test_zscal.o: in function `__ctest_zscal_i_inf_run':
/run/build/openblas/utest/test_zscal.c:62:(.text+0x5fb): undefined reference to `cblas_zscal'
/usr/lib/gcc/x86_64-unknown-linux-gnu/13.2.0/../../../../x86_64-unknown-linux-gnu/bin/ld: test_zscal.o:/run/build/openblas/utest/test_zscal.c:74: more undefined references to `cblas_zscal' follow
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:73: openblas_utest] Error 1
make[1]: Leaving directory '/run/build/openblas/utest'
make: *** [Makefile:168: tests] Error 2
```

--- a/utest/test_zscal.c
+++ b/utest/test_zscal.c
@@ -11,8 +11,8 @@
 
 CTEST(zscal, i_nan)
 {
-    int N=9;
-    int incX=1;
+    blasint N=9;
+    blasint incX=1;
     double i[] = {0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1 };
     double nan[] = {NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0};
     BLASFUNC(zscal)(&N, i, nan, &incX);
@@ -24,8 +24,8 @@ CTEST(zscal, i_nan)
 
 CTEST(zscal, i_nan_inc_2)
 {
-    int N=9;
-    int incX=1;
+    blasint N=9;
+    blasint incX=1;
     double i[] = {0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1 };
     double nan[] = {NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0,
                     NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0};
@@ -38,8 +38,8 @@ CTEST(zscal, i_nan_inc_2)
 
 CTEST(zscal, nan_i)
 {
-    int N=9;
-    int incX=1;
+    blasint N=9;
+    blasint incX=1;
     double i[] = {0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1 };
     double nan[] = {NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0};
     BLASFUNC(zscal)(&N, nan, i, &incX);
@@ -51,8 +51,8 @@ CTEST(zscal, nan_i)
 
 CTEST(zscal, nan_i_inc_2)
 {
-    int N=9;
-    int incX=1;
+    blasint N=9;
+    blasint incX=1;
     double i[] = {0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1,
                   0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1 };
     double nan[] = {NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0};
@@ -65,8 +65,8 @@ CTEST(zscal, nan_i_inc_2)
 
 CTEST(zscal, i_inf)
 {
-    int N=9;
-    int incX=1;
+    blasint N=9;
+    blasint incX=1;
     double i[] = {0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1 };
     double inf[] = {INFINITY, 0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0};
     BLASFUNC(zscal)(&N, i, inf, &incX);
@@ -78,8 +78,8 @@ CTEST(zscal, i_inf)
 
 CTEST(zscal, i_inf_inc_2)
 {
-    int N=9;
-    int incX=2;
+    blasint N=9;
+    blasint incX=2;
     double i[] = {0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1 };
     double inf[] = {INFINITY, 0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0,
                     INFINITY, 0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0};
@@ -92,8 +92,8 @@ CTEST(zscal, i_inf_inc_2)
 
 CTEST(zscal, inf_i)
 {
-    int N=9;
-    int incX=1;
+    blasint N=9;
+    blasint incX=1;
     double i[] = {0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1 };
     double inf[] = {INFINITY, 0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0};
     BLASFUNC(zscal)(&N, inf, i, &incX);
@@ -105,8 +105,8 @@ CTEST(zscal, inf_i)
 
 CTEST(zscal, inf_i_inc_2)
 {
-    int N=9;
-    int incX=2;
+    blasint N=9;
+    blasint incX=2;
     double i[] = {0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1,
                   0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1 };
     double inf[] = {INFINITY, 0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0};

--- a/utest/test_zscal.c
+++ b/utest/test_zscal.c
@@ -13,7 +13,7 @@ CTEST(zscal, i_nan)
 {
     double i[] = {0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1 };
     double nan[] = {NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0};
-    BLASFUNC(zscal)(9, i, &nan, 1);
+    BLASFUNC(zscal)(9, i, nan, 1);
     ASSERT_TRUE(isnan(nan[0]));
     ASSERT_TRUE(isnan(nan[1]));
     ASSERT_TRUE(isnan(nan[16]));
@@ -25,7 +25,7 @@ CTEST(zscal, i_nan_inc_2)
     double i[] = {0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1 };
     double nan[] = {NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0,
                     NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0};
-    BLASFUNC(zscal)(9, i, &nan, 2);
+    BLASFUNC(zscal)(9, i, nan, 2);
     ASSERT_TRUE(isnan(nan[0]));
     ASSERT_TRUE(isnan(nan[1]));
     ASSERT_TRUE(isnan(nan[16]));
@@ -36,7 +36,7 @@ CTEST(zscal, nan_i)
 {
     double i[] = {0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1 };
     double nan[] = {NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0};
-    BLASFUNC(zscal)(9, &nan, &i, 1);
+    BLASFUNC(zscal)(9, nan, i, 1);
     ASSERT_TRUE(isnan(i[0]));
     ASSERT_TRUE(isnan(i[1]));
     ASSERT_TRUE(isnan(i[16]));
@@ -48,7 +48,7 @@ CTEST(zscal, nan_i_inc_2)
     double i[] = {0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1,
                   0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1 };
     double nan[] = {NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0};
-    BLASFUNC(zscal)(9, &nan, &i, 2);
+    BLASFUNC(zscal)(9, nan, i, 2);
     ASSERT_TRUE(isnan(i[0]));
     ASSERT_TRUE(isnan(i[1]));
     ASSERT_TRUE(isnan(i[16]));
@@ -59,7 +59,7 @@ CTEST(zscal, i_inf)
 {
     double i[] = {0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1 };
     double inf[] = {INFINITY, 0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0};
-    BLASFUNC(zscal)(9, i, &inf, 1);
+    BLASFUNC(zscal)(9, i, inf, 1);
     ASSERT_TRUE(isnan(inf[0]));
     ASSERT_TRUE(isinf(inf[1]));
     ASSERT_TRUE(isnan(inf[16]));
@@ -71,7 +71,7 @@ CTEST(zscal, i_inf_inc_2)
     double i[] = {0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1 };
     double inf[] = {INFINITY, 0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0,
                     INFINITY, 0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0};
-    BLASFUNC(zscal)(9, i, &inf, 2);
+    BLASFUNC(zscal)(9, i, inf, 2);
     ASSERT_TRUE(isnan(inf[0]));
     ASSERT_TRUE(isinf(inf[1]));
     ASSERT_TRUE(isnan(inf[16]));
@@ -82,7 +82,7 @@ CTEST(zscal, inf_i)
 {
     double i[] = {0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1 };
     double inf[] = {INFINITY, 0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0};
-    BLASFUNC(zscal)(9, &inf, &i, 1);
+    BLASFUNC(zscal)(9, inf, i, 1);
     ASSERT_TRUE(isnan(i[0]));
     ASSERT_TRUE(isinf(i[1]));
     ASSERT_TRUE(isnan(i[16]));
@@ -94,7 +94,7 @@ CTEST(zscal, inf_i_inc_2)
     double i[] = {0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1,
                   0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1 };
     double inf[] = {INFINITY, 0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0};
-    BLASFUNC(zscal)(9, &inf, &i, 2);
+    BLASFUNC(zscal)(9, inf, i, 2);
     ASSERT_TRUE(isnan(i[0]));
     ASSERT_TRUE(isinf(i[1]));
     ASSERT_TRUE(isnan(i[16]));

--- a/utest/test_zscal.c
+++ b/utest/test_zscal.c
@@ -11,9 +11,11 @@
 
 CTEST(zscal, i_nan)
 {
+    int N=9;
+    int incX=1;
     double i[] = {0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1 };
     double nan[] = {NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0};
-    BLASFUNC(zscal)(9, i, nan, 1);
+    BLASFUNC(zscal)(&N, i, nan, &incX);
     ASSERT_TRUE(isnan(nan[0]));
     ASSERT_TRUE(isnan(nan[1]));
     ASSERT_TRUE(isnan(nan[16]));
@@ -22,10 +24,12 @@ CTEST(zscal, i_nan)
 
 CTEST(zscal, i_nan_inc_2)
 {
+    int N=9;
+    int incX=1;
     double i[] = {0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1 };
     double nan[] = {NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0,
                     NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0};
-    BLASFUNC(zscal)(9, i, nan, 2);
+    BLASFUNC(zscal)(&N, i, nan, &incX);
     ASSERT_TRUE(isnan(nan[0]));
     ASSERT_TRUE(isnan(nan[1]));
     ASSERT_TRUE(isnan(nan[16]));
@@ -34,9 +38,11 @@ CTEST(zscal, i_nan_inc_2)
 
 CTEST(zscal, nan_i)
 {
+    int N=9;
+    int incX=1;
     double i[] = {0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1 };
     double nan[] = {NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0};
-    BLASFUNC(zscal)(9, nan, i, 1);
+    BLASFUNC(zscal)(&N, nan, i, &incX);
     ASSERT_TRUE(isnan(i[0]));
     ASSERT_TRUE(isnan(i[1]));
     ASSERT_TRUE(isnan(i[16]));
@@ -45,10 +51,12 @@ CTEST(zscal, nan_i)
 
 CTEST(zscal, nan_i_inc_2)
 {
+    int N=9;
+    int incX=1;
     double i[] = {0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1,
                   0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1 };
     double nan[] = {NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0};
-    BLASFUNC(zscal)(9, nan, i, 2);
+    BLASFUNC(zscal)(&N, nan, i, &incX);
     ASSERT_TRUE(isnan(i[0]));
     ASSERT_TRUE(isnan(i[1]));
     ASSERT_TRUE(isnan(i[16]));
@@ -57,9 +65,11 @@ CTEST(zscal, nan_i_inc_2)
 
 CTEST(zscal, i_inf)
 {
+    int N=9;
+    int incX=1;
     double i[] = {0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1 };
     double inf[] = {INFINITY, 0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0};
-    BLASFUNC(zscal)(9, i, inf, 1);
+    BLASFUNC(zscal)(&N, i, inf, &incX);
     ASSERT_TRUE(isnan(inf[0]));
     ASSERT_TRUE(isinf(inf[1]));
     ASSERT_TRUE(isnan(inf[16]));
@@ -68,10 +78,12 @@ CTEST(zscal, i_inf)
 
 CTEST(zscal, i_inf_inc_2)
 {
+    int N=9;
+    int incX=2;
     double i[] = {0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1 };
     double inf[] = {INFINITY, 0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0,
                     INFINITY, 0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0};
-    BLASFUNC(zscal)(9, i, inf, 2);
+    BLASFUNC(zscal)(&N, i, inf, &incX);
     ASSERT_TRUE(isnan(inf[0]));
     ASSERT_TRUE(isinf(inf[1]));
     ASSERT_TRUE(isnan(inf[16]));
@@ -80,9 +92,11 @@ CTEST(zscal, i_inf_inc_2)
 
 CTEST(zscal, inf_i)
 {
+    int N=9;
+    int incX=1;
     double i[] = {0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1 };
     double inf[] = {INFINITY, 0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0};
-    BLASFUNC(zscal)(9, inf, i, 1);
+    BLASFUNC(zscal)(&N, inf, i, &incX);
     ASSERT_TRUE(isnan(i[0]));
     ASSERT_TRUE(isinf(i[1]));
     ASSERT_TRUE(isnan(i[16]));
@@ -91,10 +105,12 @@ CTEST(zscal, inf_i)
 
 CTEST(zscal, inf_i_inc_2)
 {
+    int N=9;
+    int incX=2;
     double i[] = {0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1,
                   0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1 };
     double inf[] = {INFINITY, 0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0};
-    BLASFUNC(zscal)(9, inf, i, 2);
+    BLASFUNC(zscal)(&N, inf, i, &incX);
     ASSERT_TRUE(isnan(i[0]));
     ASSERT_TRUE(isinf(i[1]));
     ASSERT_TRUE(isnan(i[16]));

--- a/utest/test_zscal.c
+++ b/utest/test_zscal.c
@@ -13,7 +13,7 @@ CTEST(zscal, i_nan)
 {
     double i[] = {0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1 };
     double nan[] = {NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0};
-    cblas_zscal(9, i, &nan, 1);
+    BLASFUNC(zscal)(9, i, &nan, 1);
     ASSERT_TRUE(isnan(nan[0]));
     ASSERT_TRUE(isnan(nan[1]));
     ASSERT_TRUE(isnan(nan[16]));
@@ -25,7 +25,7 @@ CTEST(zscal, i_nan_inc_2)
     double i[] = {0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1 };
     double nan[] = {NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0,
                     NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0};
-    cblas_zscal(9, i, &nan, 2);
+    BLASFUNC(zscal)(9, i, &nan, 2);
     ASSERT_TRUE(isnan(nan[0]));
     ASSERT_TRUE(isnan(nan[1]));
     ASSERT_TRUE(isnan(nan[16]));
@@ -36,7 +36,7 @@ CTEST(zscal, nan_i)
 {
     double i[] = {0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1 };
     double nan[] = {NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0};
-    cblas_zscal(9, &nan, &i, 1);
+    BLASFUNC(zscal)(9, &nan, &i, 1);
     ASSERT_TRUE(isnan(i[0]));
     ASSERT_TRUE(isnan(i[1]));
     ASSERT_TRUE(isnan(i[16]));
@@ -48,7 +48,7 @@ CTEST(zscal, nan_i_inc_2)
     double i[] = {0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1,
                   0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1 };
     double nan[] = {NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0, NAN,0};
-    cblas_zscal(9, &nan, &i, 2);
+    BLASFUNC(zscal)(9, &nan, &i, 2);
     ASSERT_TRUE(isnan(i[0]));
     ASSERT_TRUE(isnan(i[1]));
     ASSERT_TRUE(isnan(i[16]));
@@ -59,7 +59,7 @@ CTEST(zscal, i_inf)
 {
     double i[] = {0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1 };
     double inf[] = {INFINITY, 0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0};
-    cblas_zscal(9, i, &inf, 1);
+    BLASFUNC(zscal)(9, i, &inf, 1);
     ASSERT_TRUE(isnan(inf[0]));
     ASSERT_TRUE(isinf(inf[1]));
     ASSERT_TRUE(isnan(inf[16]));
@@ -71,7 +71,7 @@ CTEST(zscal, i_inf_inc_2)
     double i[] = {0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1 };
     double inf[] = {INFINITY, 0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0,
                     INFINITY, 0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0};
-    cblas_zscal(9, i, &inf, 2);
+    BLASFUNC(zscal)(9, i, &inf, 2);
     ASSERT_TRUE(isnan(inf[0]));
     ASSERT_TRUE(isinf(inf[1]));
     ASSERT_TRUE(isnan(inf[16]));
@@ -82,7 +82,7 @@ CTEST(zscal, inf_i)
 {
     double i[] = {0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1 };
     double inf[] = {INFINITY, 0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0};
-    cblas_zscal(9, &inf, &i, 1);
+    BLASFUNC(zscal)(9, &inf, &i, 1);
     ASSERT_TRUE(isnan(i[0]));
     ASSERT_TRUE(isinf(i[1]));
     ASSERT_TRUE(isnan(i[16]));
@@ -94,7 +94,7 @@ CTEST(zscal, inf_i_inc_2)
     double i[] = {0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1,
                   0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1, 0,1 };
     double inf[] = {INFINITY, 0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0, INFINITY,0};
-    cblas_zscal(9, &inf, &i, 2);
+    BLASFUNC(zscal)(9, &inf, &i, 2);
     ASSERT_TRUE(isnan(i[0]));
     ASSERT_TRUE(isinf(i[1]));
     ASSERT_TRUE(isnan(i[16]));

